access-control-app/project.clj

```diff
@@ -1,89 +1,85 @@
   :profiles {
-    :dev {:dependencies ~(into '[[org.clojure/tools.namespace "0.2.11"]
-                                 [pjstadig/humane-test-output "0.8.1"]
-                                 [proto-repl "0.3.1"]
-                                 [ring-mock "0.1.5"]]
-                           dev-cmr-deps)
-          :jvm-opts ^:replace ["-server"]
-          :source-paths ["src" "dev" "test" "int_test"]
-          :test-paths ["test" "int_test"]
-          :injections [(require 'pjstadig.humane-test-output)
-                       (pjstadig.humane-test-output/activate!)]}
-
+    :dev {
+      :exclusions [
+        [org.clojure/tools.nrepl]]
+      :dependencies [
+        [nasa-cmr/cmr-mock-echo-app "0.1.0-SNAPSHOT"]
+        [org.clojure/tools.namespace "0.2.11"]
+        [org.clojure/tools.nrepl "0.2.13"]
+        [pjstadig/humane-test-output "0.8.3"]
+        [proto-repl "0.3.1"]
+        [ring-mock "0.1.5"]]
+      :jvm-opts ^:replace ["-server"]
+      :source-paths ["src" "dev" "test" "int_test"]
+      :test-paths ["test" "int_test"]
+      :injections [(require 'pjstadig.humane-test-output)
+                   (pjstadig.humane-test-output/activate!)]}
     ;; This profile specifically here for generating documentation. It's faster
     ;; than using the regular profile. An agent pool is being started when
     ;; using the default profile which causes the wait of 60 seconds before
     ;; allowing the JVM to shutdown since no call to shutdown-agents is made.
     ;; Generate docs with: lein generate-static (the alias makes use of the
     ;; static profile).
     :static {}
-
-    :uberjar {:main cmr.access-control.runner
-              :aot :all}
+    :uberjar {
+      :main cmr.access-control.runner
+      :aot :all}
     ;; This profile is used for linting and static analysis. To run for this
     ;; project, use `lein lint` from inside the project directory. To run for
     ;; all projects at the same time, use the same command but from the top-
     ;; level directory.
     :lint {
       :source-paths ^:replace ["src"]
       :test-paths ^:replace []
-      :plugins [[jonase/eastwood "0.2.3"]
-                [lein-ancient "0.6.10"]
-                [lein-bikeshed "0.4.1"]
-                [lein-kibit "0.1.2"]
-                [lein-shell "0.4.0"]
-                [venantius/yagni "0.1.4"]]}}
+      :plugins [
+        [jonase/eastwood "0.2.5"]
+        [lein-ancient "0.6.15"]
+        [lein-bikeshed "0.5.0"]
+        [lein-kibit "0.1.6"]
+        [venantius/yagni "0.1.4"]]}}
   :test-paths ["test" "int_test"]
   :aliases {"generate-static" ["with-profile" "static"
                                "run" "-m" "cmr.access-control.site.static" "all"]
             ;; Prints out documentation on configuration environment variables.
             "env-config-docs" ["exec" "-ep" "(do (use 'cmr.common.config) (print-all-configs-docs))"]
-            ;; Creates the checkouts directory to the local projects
```

Comment:
This is useful for anyone try to work on just the access-control project. I think we need to put this back in (and the cmr-deps and dev-cmr-deps need to be added back to support it).

Reply:
Ah, I guess the `git grep` I did wasn't enough :-/

We can discuss this later, but this seems like an odd place to but some crucial functionality ... is there are reason it's just on `access-control`? Is this app worked on alone, more than any others?

Maybe this would find a better home somewhere more generally useful?

Reply:
Also, by putting this change back in (which I'm doing right now!), we loose the ability to check the deps versions using `lein ancient`, due to the way they are dynamically constructed ...

Reply:
Okay, I added the checkouts alias back in, but I did it a bit differently -- I used the approach that `dev-system` uses (it's cleaner).